### PR TITLE
build: hash and create ".sha256sum" for binary built

### PR DESCRIPTION
**Pull request by jesec**

### CODE REVIEW

File: lib/build.ts [hunk 1]
```diff
@@ -166,6 +166,26 @@ async function compile(nodeVersion: string, targetArch: string) {
   return compileOnUnix(nodeVersion, targetArch);
 }
 
+async function hash(filePath: string): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    const resultHash = crypto.createHash('sha256');
+    const input = fs.createReadStream(filePath);
+
+    input.on('error', (e) => {
```
Comment (robertsLando) [inline lib/build.ts:174]: Would use `once` here
```suggestion
    input.once('error', (e) => {
```

File: lib/build.ts [hunk 2]
```diff
@@ -166,6 +166,26 @@ async function compile(nodeVersion: string, targetArch: string) {
...
+      reject(e);
+    });
+
+    input.on('readable', () => {
+      const data = input.read();
+      if (data) {
+        resultHash.update(data);
+      } else {
+        resolve(resultHash.digest('hex'));
```
Comment (robertsLando) [inline lib/build.ts:183]: Should we also destroy the stream here?

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: lib/build.ts (hunk 2 above)
Comment (jesec) [reply lib/build.ts:183]: There is no need to. The process exits shortly after anyways.

File: lib/build.ts (hunk 1 above)
Comment (jesec) [reply lib/build.ts:174]: We don't want unhandled exceptions. It is more common to use `on`.

**Review by robertsLando — APPROVED** [review]

**Pull request merged**